sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1263,6 +1250,46 @@ def test_gaussian_mixture_setting_best_params():
         assert hasattr(gmm, attr)
 
 
+@pytest.mark.parametrize(
+    "init_params", ["random", "random_from_data", "k-means++", "kmeans"]
+)
+def test_check_non_unique_means_initialisation(init_params):
+    # Check that all initialisations provide unique starting means
```

Comment:
I'm not sure what this test is supposed to check. Is it that different init_params lead to all different init means ? I don't think we can ensure that. It's unlikely, but it might happen for some random seed.

I think it would be better to check that general properties hold for each init_params.

Reply:
I think the purpose is rather to ensure that the means within each initialization are different, which makes a bit more sense.  Do you still think it's better to remove it? I could change the comment to be more explicit and add your suggested general properties test as a new test.

Reply:
(Sorry, I just noticed that you answered).

> I think the purpose is rather to ensure that the means within each initialization are different

That would be acceptable to test indeed. But that's not what the current code is doing.

>  I could change the comment to be more explicit and add your suggested general properties test as a new test.

Sure. I think it can be done in this same test.